File: 프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque
def solution(order):
    answer = []
    stack = []
    order = deque(order)
    for i in range(1, len(order)+1):
        stack.append(i)
        while order:
            if stack and stack[-1] == order[0]:
                answer.append(order.popleft())
                stack.pop()
            else:
                break

    return len(answer)
print(solution([1,2,3,4,5]))

File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
  let result = 0;
  const stack = [];

  for (let i = 1; i <= order.length; i++) {
    stack.push(i);

    while (stack.length !== 0 && stack.at(-1) === order[result]) {
      stack.pop();
      result++;
    }
  }

  return result;
}